packages/common/src/extensions/slickRowBasedEdit.ts

```diff
@@ -0,0 +1,596 @@
+          onBeforePasteCell: (e: SlickEventData<any>, args: OnEventArgs) => {
+            let userResult = true;
+            if (existingBeforePasteCellHandler) {
+              userResult = existingBeforePasteCellHandler(e, args);
+
+              if (userResult === false) {
+                return false;
+              }
+            }
+
+            const item = dataView.getItem(args.row);
+            const idProperty =
+              this._grid.getOptions().datasetIdPropertyName ?? 'id';
+
+            if (this._editedRows.has(item[idProperty]) && userResult === true) {
+              return true;
+            }
+
+            return false;
+          },
+        },
+      });
+    }
+
+    const originalGetItemMetadata = dataView.getItemMetadata;
+    dataView.getItemMetadata = this.updateItemMetadata(
+      originalGetItemMetadata?.bind?.(dataView)
+    );
+    this._eventHandler.subscribe(
+      this._grid.onSetOptions,
+      this.optionsUpdatedHandler.bind(this)
+    );
+    this._eventHandler.subscribe(
+      dataView.onRowsOrCountChanged,
+      this.handleAllRowRerender.bind(this)
+    );
+
+    this._grid.invalidate();
+    this._grid.render();
+  }
+
+  destroy() {
+    this.dispose();
+  }
+
+  /** Dispose (destroy) the SlickGrid 3rd party plugin */
+  dispose() {
+    this._eventHandler?.unsubscribeAll();
+    this.pubSubService?.unsubscribeAll();
+  }
+
+  create(
+    columnDefinitions: Column[],
+    gridOptions: GridOption
+  ): SlickRowBasedEdit | null {
+    this._addonOptions = {
+      ...this._defaults,
+      ...gridOptions.rowBasedEditOptions,
+    } as RowBasedEditOptions;
+    if (Array.isArray(columnDefinitions) && gridOptions) {
+      const selectionColumn: Column = this.getColumnDefinition();
+      // add new action column unless it was already added
+      if (!columnDefinitions.some((col) => col.id === selectionColumn.id)) {
+        // column index position in the grid
+        const columnPosition =
+          gridOptions?.rowBasedEditOptions?.columnIndexPosition ?? -1;
+        if (columnPosition === -1) {
+          columnDefinitions.push(selectionColumn);
+        } else if (
+          columnPosition > 0 &&
+          columnPosition < columnDefinitions.length
+        ) {
+          columnDefinitions.splice(columnPosition, 0, selectionColumn);
+        } else {
+          columnDefinitions.unshift(selectionColumn);
+        }
+        this.pubSubService.publish(`onPluginColumnsChanged`, {
+          columns: columnDefinitions,
+          pluginName: this.pluginName,
+        });
+      }
+    }
+    return this;
+  }
+
+  getColumnDefinition(): Column {
+    const columnId = String(
+      this._addonOptions?.columnId ?? this._defaults.columnId
+    );
+
+    return {
+      id: columnId,
+      name: this._addonOptions?.actionsColumnLabel,
+      field: 'action',
+      minWidth: 70,
+      width: 75,
+      maxWidth: 75,
+      excludeFromExport: true,
```

Comment:
good catch. actually I think instead of defining width and others we should accept `actionColumnConfig: Partial<Column<any>> and spread that over the previously applied. this way users can override and extens whatever they want